### PR TITLE
Removed params.config_profile_name, added params.hostnames

### PR DESCRIPTION
**Pull request by ewels**
`params.config_profile_name` didn't seem to add anything beyond what we already had in the profile name, so I got rid of it.

I also added a new groovy map on the main config file called `params.hostnames`... Why? All will soon become clear...

### CODE REVIEW

File: conf/crick.config [hunk 1]
```diff
@@ -1,8 +1,7 @@
 //Profile config names for nf-core/configs
 params {
-  config_profile_name = 'CRICK'
   config_profile_description = 'The Francis Crick Institute CAMP HPC cluster profile provided by nf-core/configs.'
-  config_profile_contact = 'Harshil Patel (@drpatelh )'
```
Comment (drpatelh) [inline conf/crick.config]: Ive been meaning to remove that bloody space for ages :+1: :laughing:

**Comment by ewels** [comment]
x-ref: https://github.com/nf-core/tools/pull/278

File: nfcore_custom.config [hunk 2]
```diff
@@ -23,9 +23,17 @@ profiles {
...
   uppmax       { includeConfig "${params.custom_config_base}/conf/uppmax.config" }
   uzh          { includeConfig "${params.custom_config_base}/conf/uzh.config" }
 }
+
+// If user hostnames contain one of these substring and they are
+// not running the associated profile, it will trigger a warning message
+// Should be defined here for all profiles (not within profile config)
+params {
+  // This is a groovy map, not a nextflow parameter set
+  hostnames = [
```
Comment (drpatelh) [inline nfcore_custom.config]: Think Ive figured out whats going on by looking at the PR in `nf-core/tools`.

We would all have to explicitly specify a `hostname` value here right e.g.  
```nextflow
params {
￼	  // This is a groovy map, not a nextflow parameter set
￼	  hostnames = [
              uppmax: ['.uppmax.uu.se'],
              crick: ['.thecrick.org']
￼	  ]
￼	}


```

Comment (drpatelh) [reply nfcore_custom.config]: Everything else looks good to me.

Comment (ewels) [reply nfcore_custom.config]: Yes - that is correct :) So more PRs needed to add all of those, but seems to work nicely from my initial testing. Also needs PRs to every pipeline to add the function that uses this. Now it's being added to the template it should slowly propagate out though..

Comment (ewels) [reply nfcore_custom.config]: This is what I get with test data on the dev rnaseq pipeline now:
image

**Pull request merged**